c10/core/DispatchKey.h

```diff
@@ -420,10 +534,65 @@ C10_API c10::DispatchKey parseDispatchKey(const std::string& k);
 // torch::dispatch(torch::kCPU, ...) is also valid.
 constexpr DispatchKey kAutograd = DispatchKey::Autograd;
 
-// Check if a DispatchKey is an alias mapping to other runtime keys.
-inline bool isAliasDispatchKey(DispatchKey k) {
-  return k > DispatchKey::NumDispatchKeys && k <= DispatchKey::EndOfAliasKeys;
+// This function relies on the invariant that the dispatch keys between
+// StartOfDenseBackends and EndOfAutogradBackends are ordered by backend
+// in the same order as `BackendBit`.
+// TODO: statically assert this invariant.
+constexpr BackendBit toBackendBit(DispatchKey k) {
```

Comment:
It'd need to be updated whenever we add a new type of per-backend-functionality (e.g. if we make `Autocast` or `SparseCsr` into a per-backend functionality).

In practice when we do that, I'd probably just grep for all instances of `DispatchKey::EndOfAutogradBackends` (or a similar key) to know where to change things

Reply:
As discussed in other places, this should be documented!

Reply:
Added the assert for the invariant (I actually added is as a unit test since I thought the full set of static asserts would be unwieldy). And added some more comments/documentation.